src/dbnode/storage/namespace_readers.go

```diff
@@ -380,3 +457,31 @@ func (m *namespaceReaderManager) tickWithThreshold(threshold int) {
 		m.openReaders[key] = elem
 	}
 }
+
+// UpdateOpenLease() implements block.Leaser.
+func (m *namespaceReaderManager) UpdateOpenLease(
+	descriptor block.LeaseDescriptor,
+	state block.LeaseState,
+) (block.UpdateOpenLeaseResult, error) {
+	if !m.namespace.ID().Equal(descriptor.Namespace) {
+		return block.NoOpenLease, nil
+	}
+
+	m.Lock()
```

Comment:
Any reason to manually Lock and Unlock here instead of defer?

Reply:
I guess it doesn't really matter much since there aren't any early returns

Reply:
Not really. I'll change it to a defer to avoid bugs in the future.